### PR TITLE
Add numpy-allocator

### PR DESCRIPTION
**Pull request by hmaarrfk**
@jakirkham did you want to help maintain? I noticed you were active in the discussion referenced below.

@eliaskoromilas do you have any interest in helping maintain conda-forge packages?

https://github.com/numpy/numpy/issues/20193#issuecomment-963631602

Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by eliaskoromilas** [comment]
> @eliaskoromilas do you have any interest in helping maintain conda-forge packages?

I could help with this one. I will check the windows issue. Maybe it's time to publish macos and windows wheels, too.

**Comment by hmaarrfk** [comment]
ah you never published windows wheels. Yeah, i think we can "fix" those two platforms later. good to know it isn't just a "me" thing that the compilation didn't work.

**Comment by hmaarrfk** [comment]
I'm going to leave it here to see if jakirkham wants anything (maybe regarding the naming and whatnot) but otherwise, i skipped the winodws builds so i think we are ok!

File: recipes/numpy-allocator/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  imports:
+    - numpy_allocator
+  requires:
+    - pip
+  commands:
+    - pip check
+
+about:
+  home: https://github.com/inaccel/numpy-allocator
+  summary: Customize Memory Allocators for NumPy
```
Comment (eliaskoromilas) [inline recipes/numpy-allocator/meta.yaml:45]:
```suggestion
  summary: Customize Memory Allocators in NumPy
```

**Comment by eliaskoromilas** [comment]
I change some logic to also support PyPy and I will publish today a new version that includes the windows fix.

**Comment by hmaarrfk** [comment]
sounds good. thank you for this. I was going to try to attempt to do the same to learn ho wit all worked. Out of curiosity. does your package make it so that when you use an aligned allocator, the python GIL is acquired again during allocation? or does everything happen in the C world without the GIL?

**Comment by jakirkham** [comment]
> did you want to help maintain? I noticed you were active in the discussion referenced below.

No. Thanks for asking though

**Comment by jakirkham** [comment]
Thank you both! 🙏

**Pull request merged**